connect to routers even if we are decomissioned

majestrate opened the pull request:
fixes #1904

Commit:
connect to routers even if we are decomissioned

jagerman commented:
Confirmed on testnet that this works.

While testing this I noticed something odd that should also be addressed: unregistered routers are still performing tests -- these tests always fail, however, because the testee rejects a connection from an unregistered router.  We should just stop testing completely if we aren't active or decomissioned.

majestrate committed:
if we are decommissioned or deregistered, do not test other routers so we do not spam them.

disambiguiate error message to distinguish between decomissioned and deregistered.

majestrate committed:
publish our rc out to the network when we regenerate them

jagerman commented on llarp/router/router.cpp:
```diff
@@ -889,15 +908,22 @@ namespace llarp
     {
       LogInfo("regenerating RC");
-      if (!UpdateOurRC(false))
-        LogError("Failed to update our RC");
```
I think we should keep this in an `else` so that we know from log output when something getting wrong.

majestrate committed:
add gossip info to systemd status

* adds next and last gossip datetimes
* adds a few things for time points, like ostream operator overloads for time point

majestrate committed:
format systemd status as time deltas from now

jagerman commented on llarp/util/time.cpp:
```diff
@@ -74,4 +78,27 @@ namespace llarp
+      std::stringstream ss;
+      const auto dlt = TimePoint_t::clock::now() - at;
+      if (dlt > 0s)
+        ss << std::chrono::duration_cast<Duration_t>(TimePoint_t::clock::now() - at) << " ago ";
```
```suggestion
        ss << std::chrono::duration_cast<Duration_t>(dlt) << " ago ";
```

jagerman commented on llarp/util/time.cpp:
```diff
@@ -74,4 +78,27 @@ namespace llarp
+      if (dlt > 0s)
+        ss << std::chrono::duration_cast<Duration_t>(TimePoint_t::clock::now() - at) << " ago ";
+      else if (dlt < 0s)
+        ss << "in " << std::chrono::duration_cast<Duration_t>(TimePoint_t::clock::now() - at);
```
I think this is supposed to be negated, i.e. `at - now`, or alternatively:
```suggestion
        ss << "in " << std::chrono::duration_cast<Duration_t>(-dlt);
```

majestrate committed:
set source ip on service nodes for outbound link to not use all interfaces

majestrate committed:
footcannon prevention: check for invalid address family.

throw if we pass in a bogus af value when getting a sockaddr for all interfaces

jagerman commented on llarp/link/server.cpp:
```diff
@@ -139,28 +140,60 @@ namespace llarp
           RecvFrom(from, std::move(pkt));
         });
-
+    std::string _ifname{ifname};
```
If it needs an unconditional copy, might as well take it by value rather than `const&`.
